### DISCUSSION TITLE
how to use as midi in to serial 1 for use with wav trigger

### DISCUSSION THREAD

**Discussion by jmartens2** [tttapa/Control-Surface#471] [General]
hi. im not sure how to explain this because im a noobie to most things with communication.
i would like to know if there are any examples that could help teach me about using midi out from a roland drum machine
to mega 2560 ardino board. i would like to be able to output serial port 1 to sparkfun wav trigger to be able to trigger wav samples.
ive spent several hours today asking ms. google and everything is confusing as heck. any help in finding examples, tutorials, and sketches would be helpful.
thank you for your time if you read this!

**Comment by tttapa** [comment]
It's not entirely clear what you're after. Control Surface can talk MIDI with the WAV trigger and the Roland drum, see https://tttapa.github.io/Control-Surface-doc/new-input/Doxygen/df/df5/md_pages_FAQ.html#faq-midi and the examples linked to in that FAQ.

It's not clear to me what the purpose of the Arduino is. Why not connect the drum to the WAV trigger directly?

Given that this is question is not about the Control Surface library directly, you might get more answers on the [Arduino forum](https://forum.arduino.cc/).

**Comment by jmartens2** [comment]
Hi I’m quite new to all this and I’m very sorry for confusion. I will check the faq. I was trying to figure if I needed a buffer between the two. Thank you so much for that bit of info.

**Comment by jmartens2** [comment]
well id like to eventually make a aduino edrum interface. the mega seems as if it has enough analog inputs and memory to start with. many of the projects over the web are confusing and they dont work when i try them. so i would like to re-frame my query in this simpler version. can i use your software to accomplish this task? as always thank you for your hard work and time.
